pysap/numerics/opt.py

```diff
@@ -0,0 +1,95 @@
+import time
+from modopt.opt.algorithms import Condat, ForwardBackward
+from modopt.opt.cost import costObj
+from modopt.opt.linear import Identity
+from pysap.utils import fista_logo, condatvu_logo
+
+
+def opt_fb(gradient_op, prox_op, **kwargs):
+    # FIXME find zero initialization with proper size if no x_init given
+    x_init = kwargs.get('x_init')
+    beta_param = kwargs.get('beta_param', 1.0)
+    lambda_param = kwargs.get('lambda_param', 1.0)
+    beta_update = kwargs.get('beta_update', None)
+    lambda_update = kwargs.get('lambda_update', 'fista')
+    cost = kwargs.get('cost', None)
+    verbose = kwargs.get('verbose', 1)
+    max_iter = kwargs.get('max_iter', 150)
+
+    if verbose:
+        print(fista_logo())
+        print(" - Lipschitz constant: ", gradient_op.spec_rad)
+        print(" - data: ", gradient_op.obs_data.shape)
+        print(" - max iterations: ", max_iter)
+        print(" - iterate variable shape: ", x_init.shape)
+        print("-" * 40)
+
+    start = time.clock()
+
+    opt = ForwardBackward(
+        x=x_init,
+        grad=gradient_op,
+        prox=prox_op,
+        cost=cost,
+        beta_param=beta_param,
+        lambda_param=lambda_param,
+        beta_update=beta_update,
+        lambda_update=lambda_update,
+        auto_iterate=False)
+    opt.iterate(max_iter)
```

Comment:
Maybe we can add the dual gap for this formulation as a criterion

Reply:
It will be done in later PR's, right now I just want to give an overview of the architecture